### ISSUE TITLE
ROL build failure in new CUDA 9.2 build on 'white'

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#3284]
CC: @trilinos/rol, @fryeguy52, @rppawlo (Trilinos Nonlinear Algorithms Product Lead)

## Description

The source file `packages/rol/example/PDE-OPT/TEST/test_06.cpp` fails to compile in the new `Trilinos-atdm-white-ride-cuda-9.2-debug-pt` on 'white' as shown [here](https://testing-vm.sandia.gov/cdash/viewBuildError.php?buildid=3820252) which shows the build failure:

```
/home/jenkins/white/workspace/Trilinos-atdm-white-ride-cuda-9.2-debug-pt/SRC_AND_BUILD/Trilinos/packages/rol/example/PDE-OPT/TEST/../TOOLS/template_tools.hpp:113:144: error: expansion pattern ‘Index’ contains no argument packs
 template<class Container,class Index=index_t<Container>>
                                                                                                                                                ^  
/home/jenkins/white/workspace/Trilinos-atdm-white-ride-cuda-9.2-debug-pt/SRC_AND_BUILD/Trilinos/packages/rol/example/PDE-OPT/TEST/../TOOLS/template_tools.hpp:113:147: error: template argument 4 is invalid
 template<class Container,class Index=index_t<Container>>
                                                                                                                                                   ^
/home/jenkins/white/workspace/Trilinos-atdm-white-ride-cuda-9.2-debug-pt/SRC_AND_BUILD/Trilinos/packages/rol/example/PDE-OPT/TEST/../TOOLS/template_tools.hpp:113:180: error: template argument 1 is invalid
 template<class Container,class Index=index_t<Container>>
```

This is the only build failure for this new build as shown [here](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&parentid=3820222).

It is important to get this fixed pretty quickly since ROL is used in that Trilinos build for SPARC and we will be adding ROL to all of the ATDM Trilinos builds shortly (see [TRIL-212](https://software-sandbox.sandia.gov/jira/browse/TRIL-212)).

FYI: Once this build error is fixed, a new issue will be submitted for the timing-out ROL tests shown [here](https://testing-vm.sandia.gov/cdash/viewTest.php?onlyfailed&buildid=3820252).

## Steps to Reproduce

Instructions are at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

Specific instructions for 'white' are at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#ridewhite

Once Trilinos is cloned on 'white' and the 'develop' branch is checked out, one should be able to reproduce using the commands:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh cuda-9.2-debug

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_ROL=ON \
  $TRILINOS_DIR

$ make NP=16
```

**Comment by bartlettroscoe** [comment]
@dridzal said:

> We have issued a PR that should take care of the build failure. 

Thanks @dridzal!  Just add a comment here once that PR (#3297) is merged to 'develop' and then we will watch the build the next day on CDash.

**Comment by bartlettroscoe** [comment]
The merge of PR #3297 looks to have fixed this build error in this build as shown [here](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&parentid=3834831) this morning (note the `-1` subscript by the `0` in the `Build Error` column in the `ROL` row).

Closing as complete.
